tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -66,6 +66,10 @@ static void uart_async_test_init(void)
 	k_sem_reset(&rx_buf_coherency);
 	k_sem_reset(&rx_buf_released);
 	k_sem_reset(&rx_disabled);
+#ifdef DUT_IS_UART_EMUL
+	uart_emul_flush_rx_data(uart_dev);
+	uart_emul_flush_tx_data(uart_dev);
+#endif /* DUT_IS_UART_EMUL */
```

Comment:
The emulated driver should behave like a real uart, I don't expect any changes should be needed for the uart test suite. as this is a clear indication that the emulated uart is not behaving as expected :)

On real hardware, the following calls should clear the rx and tx data respectively
```
	uart_rx_disable(uart_dev);
	uart_tx_abort(uart_dev);
```

Reply:
I removed this, was no longer necessary anyways.